sdk/keyvault/azure-keyvault-administration/samples/access_control_operations.py

```diff
@@ -0,0 +1,93 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+
+from azure.keyvault.administration import (
+    KeyVaultAccessControlClient,
+    KeyVaultDataAction,
+    KeyVaultPermission,
+    KeyVaultRoleScope,
+)
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+# ----------------------------------------------------------------------------------------------------------
+# Prerequisites:
+# 1. A managed HSM (https://docs.microsoft.com/azure/key-vault/managed-hsm/quick-create-cli)
+#
+# 2. azure-keyvault-administration and azure-identity libraries (pip install these)
+#
+# 3. Set Environment variables AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET, VAULT_URL
+#    (See https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/keyvault/azure-keyvault-administration#authenticate-the-client)
+#
+# ----------------------------------------------------------------------------------------------------------
+# Sample - demonstrates role definition and assignment operations for Managed HSM
+#
+# 1. Create a role definition (set_role_definition)
+#
+# 2. Update a role definition (set_role_definition)
+#
+# 3. Create a role assignment (create_role_assignment)
+#
+# 4. Delete a role assignment (delete_role_assignment)
+#
+# 5. Delete a role definition (delete_role_definition)
+# ----------------------------------------------------------------------------------------------------------
+
+VAULT_URL = os.environ["VAULT_URL"]
+
+# Instantiate an access control client that will be used to call the service.
+# Notice that the client is using default Azure credentials.
+# To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
```

Comment:
I worry that prescribing an authentication method could create confusion by suggesting the authentiation method matters, or conflating the principal assigning the role and the principal receiving the role assignment. You really need only AZURE_CLIENT_ID, to identify the principal receiving the assignment, and any of DefaultAzureCredential's many, so many!, possibilities will do for authenticating the client.

Reply:
Yeah, this is another thing that I borrowed from our existing samples but felt could use some reworking. I'll change this